smartsim/experiment.py

```diff
@@ -232,8 +232,10 @@ def execute_dispatch(generator: Generator, job: Job, idx: int) -> LaunchedJobID:
                 launch_config = dispatch.create_new_launcher_configuration(
                     for_experiment=self, with_arguments=args
                 )
-            job_execution_path = self._generate(generator, job, idx)
-            id_ = launch_config.start(exe, job_execution_path, env)
+            # Generate the job directory and return the generated job path
+            ret = self._generate(generator, job, idx)
+            job_execution_path, out, err = ret
```

Comment:
style nit if you want to save a line:
```suggestion
            job_execution_path, out, err = self._generate(generator, job, idx)
```

Reply:
awesome catch, always down to save a line